### PR TITLE
add project

### PR DESCRIPTION
**Pull request by AndrewOsypenko**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <div>
+    <h1>Todo List</h1>
+    <AddTodo />
+    <hr>
+    <todoList />
+  </div>
+</template>
+<script>
+import TodoList from './components/TodoList.vue'
```
Comment (akoidan) [inline src/App.vue:10]: Can we use absolute path?

File: src/App.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <div>
+    <h1>Todo List</h1>
+    <AddTodo />
+    <hr>
```
Comment (akoidan) [inline src/App.vue:5]: Do not use not enclosing tags, even if w3c allows it.

File: src/App.vue [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <todoList />
```
Comment (akoidan) [inline src/App.vue:6]: Think if you lead the development if this project, how can you enforce name consistency for components? `<AddTodo />` `  <todoList />`

File: src/components/TodoList.vue [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <div>
+    <ul>
+      <TodoItem
+          v-for="todo in getAllTodos" :key="todo.id"
```
Comment (akoidan) [inline src/components/TodoList.vue]: What's the rule for number elements in line? Why this questions is important at all?

File: src/components/todo.vue [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <li style="
+    justify-content: space-around;
```
Comment (akoidan) [inline src/components/todo.vue:4]: move to css

File: src/components/todo.vue [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    border: 1px solid #ccc;
+    padding: .5rem 2rem;
+    margin-bottom: 1rem;
+">
+    <span>
+      {{ todo.title }}
+    </span>
+    <input type="text" v-model="newTitle" placeholder="New title name">
+    <button @click="updateItem">Edit</button>
+    <button @click="removeTodo(todo._id)">&times;</button>
```
Comment (akoidan) [inline src/components/todo.vue]: Does frontend use `_id` model?

File: src/store/modules/todo.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+
+export default {
+    actions: {
+        async getAll(ctx) {
+            axios.get("http://127.0.0.1:3000")
```
Comment (akoidan) [inline src/store/modules/todo.js]: again, async/await

Comment (akoidan) [inline src/store/modules/todo.js]: What's address is doing here?

File: src/store/modules/todo.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        },
+        async addTodo(ctx, title) {
+            axios.post("http://127.0.0.1:3000", {title})
+                .then(response => {
+                    ctx.dispatch('getAll');
+                });
+        },
+        async updateTodo(ctx, data) {
+            console.log(data)
+            axios.patch(`http://127.0.0.1:3000/${data.id}`, {title : data.title})
```
Comment (akoidan) [inline src/store/modules/todo.js]: I don't like the idea of store being coupled with rest api. Despite vue guide suggests this pattern, how do you unit test actions if needed?

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        path: path.resolve(__dirname, 'dist')
+    },
+    module: {
+        rules: [
+            {
+                test: /\.vue$/,
+                loader: 'vue-loader'
+            },
+            {
+                test: /\.css$/,
```
Comment (akoidan) [inline webpack.config.js:23]: Do we use css?

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
+const path = require('path')
+const VueLoaderPlugin = require('vue-loader/lib/plugin')
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
```
Comment (akoidan) [inline webpack.config.js:6]: Good minimal config. Nicely done!

File: src/components/TodoList.vue [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    TodoItem
+  },
+  computed: {
+    ...mapGetters(["getAllTodos"])
+  },
+  methods: {
+    ...mapActions(["getAll"]),
+  },
+  async mounted() {
+    await this.getAll();
```
Comment (akoidan) [inline src/components/TodoList.vue:29]: Do you refresh todos on every update?

File: src/App.vue (hunk 1 above)
Comment (akoidan) [reply src/App.vue:10]: This happens to people who don't use semicolons in javascript.
```typescript
(this as any).start = function() {
  console.log('start')
}

(this as any).finish = function() {
  console.log('finish');
}

this.finish()
```

what's the output?

Comment (AndrewOsypenko) [reply src/App.vue:10]: it should be 'finish' because ASI set semicolon after {} and  console.log('start')

File: src/App.vue (hunk 3 above)
Comment (AndrewOsypenko) [reply src/App.vue:6]: My bad. No, its not acceptable names for components. I think, it should be called like "addTaskComponent" and "tasksListComponent"

File: src/App.vue (hunk 1 above)
Comment (akoidan) [reply src/App.vue:10]: stdout of the code above:
```
start
TypeError: Cannot set property 'finish' of undefined
```

File: src/App.vue (hunk 3 above)
Comment (akoidan) [reply src/App.vue:6]: I mean `camelCase` vs `PascalCase`. How do you enforce so everyone use a specific one?

**Pull request closed without merging**